File: data/readable_format/cpp/file1280.cpp
#include <iostream>
#include <vector>
#include <cstring>

// Function 'f' translated from Fortran
int f(int x) {
    return 2*x + 1;
}

// Class 'Test1' translated from Fortran module
class Test1 {
public:
    // Method 'test2' translated from Fortran function
    static std::vector<char> test2(int x) {
        int len_r = f(x); // Using function 'f' to determine the size
        std::vector<char> r(len_r);
        for(int i = 0; i < len_r; ++i) {
            r[i] = static_cast<char>((i % 32) + '@' + 1);
        }
        return r;
    }
};

int main() {
    // The 'main' logic from the Fortran program
    std::vector<char> chr = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U'};
    auto test2_result = Test1::test2(10);

    if(test2_result.size() != 21) {
        std::cerr << "STOP 1" << std::endl;
        return 1;
    }

    bool mismatch = false;
    for(size_t i = 0; i < chr.size(); ++i) {
        if(test2_result[i] != chr[i]) {
            mismatch = true;
            break;
        }
    }

    if(mismatch) {
        std::cerr << "STOP 2" << std::endl;
        return 2;
    }

    std::cout << "Program finished successfully." << std::endl;
    return 0;
}

File: data/readable_format/fortran/file1280.f90
program test_program
  use, intrinsic :: iso_fortran_env, only: stdout => output_unit
  implicit none
  character(len=:), allocatable :: chr
  character(len=21) :: expected = "ABCDEFGHIJKLMNOPQRSTU"
  logical :: test_passed
  integer :: i

  chr = test2(10) ! Call test2 to generate the string

  if (len(chr) /= 21) then
    write(stdout,*) "Length check failed."
    stop
  end if

  test_passed = .true.
  do i = 1, len(chr)
    if (chr(i:i) /= expected(i:i)) then
      test_passed = .false.
      exit
    end if
  end do

  if (.not. test_passed) then
    write(stdout,*) "Content check failed."
    stop
  else
    write(stdout,*) "All tests passed successfully."
  end if

contains

  pure function f(x) result(res)
    integer, intent(in) :: x
    integer :: res
    res = 2*x + 1
  end function f

  function test2(x) result(r)
    integer, intent(in) :: x
    character(len=:), allocatable :: r
    integer :: i, len_r

    len_r = f(x) ! Determine the length of r based on function f
    allocate(character(len=len_r) :: r) ! Allocate r with the calculated length

    do i = 1, len_r
      r(i:i) = achar(mod(i, 32) + iachar('@'))
    end do
  end function test2

end program test_program